RevenueCatUI/Templates/Components/ScreenConditionObserver.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ScreenConditionObserver.swift
+//
+//  Created by Josh Holtz on 10/27/24.
+
+import Combine
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+enum ScreenCondition {
+    case `default`, mobileLandscape, tablet, tabletLandscape, desktop
+}
+
+class ScreenConditionObserver: ObservableObject {
+
+    private enum Device {
+        case phone, tablet, desktop
+    }
```

Comment:
Same comment as https://github.com/RevenueCat/purchases-ios/pull/4414#discussion_r1819385459. Just tagging it on here for tracking purposes.